controller/src/main/java/io/pravega/controller/server/eventProcessor/requesthandlers/TruncateStreamTask.java

```diff
@@ -58,31 +59,35 @@ public TruncateStreamTask(final StreamMetadataTasks streamMetadataTasks,
         String scope = request.getScope();
         String stream = request.getStream();
 
-        return streamMetadataStore.getTruncationRecord(scope, stream, true, context, executor)
-                .thenCompose(property -> {
-                    if (!property.isUpdating()) {
-                        // if the state is TRUNCATING but the truncation record is not updating, we should reset the state to ACTIVE.
-                        return streamMetadataStore.resetStateConditionally(scope, stream, State.TRUNCATING, context, executor)
-                                .thenRun(() -> {
+        return streamMetadataStore.getVersionedState(scope, stream, context, executor)
+                .thenCompose(versionedState -> streamMetadataStore.getTruncationRecord(scope, stream, context, executor)
+                        .thenCompose(versionedMetadata -> {
+                            if (!versionedMetadata.getObject().isUpdating()) {
```

Comment:
I have the same question about the verification here.

Reply:
replied above.